V24.3.0 beta.1 release notes (#19041)
cockroachdb/docs: src/current/_includes/releases/v24.3/v24.3.0-beta.1.md

## v24.3.0-beta.1

Release Date: October 24, 2024

{% include releases/new-release-downloads-docker-image.md release=include.release %}

<h3 id="v24-3-0-beta-1-general-changes">General changes</h3>

> - The cluster setting [`diagnostics.reporting.enabled`]({% link v24.3/cluster-settings.md %}#setting-diagnostics-reporting-enabled) is now ignored if the cluster has a [Trial or Free license]({% link v24.3/licensing-faqs.md %}#obtain-a-license), or if the reporting job is unable to load any license at all. [#132257][#132257]

Reviewer: suggest changing text from 'Trial or Free license' to 'Enteprise Trial or Enterprise Free license' and linking to the same page but the `#types-of-licenses` heading, which will be updated to describe those licenses ~next week via https://github.com/cockroachdb/docs/pull/19002

> - The cluster setting [`diagnostics.reporting.enabled`]({% link v24.3/cluster-settings.md %}#setting-diagnostics-reporting-enabled) is now ignored if the cluster has a [Trial or Free license]({% link v24.3/licensing-faqs.md %}#obtain-a-license), or if the reporting job is unable to load any license at all. [#132257][#132257]

Author: ```suggestion
- The cluster setting [`diagnostics.reporting.enabled`]({% link v24.3/cluster-settings.md %}#setting-diagnostics-reporting-enabled) is now ignored if the cluster has a [{{ site.data.products.enterprise }} Trial or {{ site.data.products.enterprise }} Free license]({% link v24.3/licensing-faqs.md %}#types-of-licenses), or if the reporting job is unable to load any license at all. [#132257][#132257]
```

<h3 id="v24-3-0-beta-1-sql-language-changes">SQL language changes</h3>

- Implemented `DROP TRIGGER` statements. The `CASCADE` option for dropping a [trigger]({% link v24.3/sql-feature-support.md %}) is not supported. [#128540][#128540]
- Added support for `CREATE TRIGGER`. The `OR REPLACE` syntax is not supported. Also, [triggers]({% link v24.3/sql-feature-support.md %}) cannot be executed, so creation is a no-op. [#128540][#128540]
- [`REGIONAL BY ROW`]({% link v24.3/table-localities.md %}#regional-by-row) and [`PARTITION ALL BY`]({% link v24.3/partitioning.md %}) tables can now be inserted into under [non-`SERIALIZABLE` isolation levels]({% link v24.3/read-committed.md %}) as long as there is no `ON CONFLICT` clause in the statement. Also, [`REGIONAL BY ROW`]({% link v24.3/table-localities.md %}#regional-by-row) and [`PARTITION ALL BY`]({% link v24.3/partitioning.md %}) tables can now be updated under non-`SERIALIZABLE` isolation levels. [#129837][#129837]

> - Attempting to add [foreign keys]({% link v24.3/foreign-keys.md %}) referencing a table with [row-level TTL]({% link v24.3/row-level-ttl.md %}) enabled will generate a notice informing the user about potential impact on the row-level TTL deletion job. Similarly, a notice is generated while attempting to enable row-level TTL on a table that has inbound foreign key references. [#127935][#127935]

Reviewer 2: ```suggestion
- Attempting to add [foreign keys]({% link v24.3/foreign-key.md %}) referencing a table with [row-level TTL]({% link v24.3/row-level-ttl.md %}) enabled will generate a notice informing the user about potential impact on the row-level TTL deletion job. Similarly, a notice is generated while attempting to enable row-level TTL on a table that has inbound foreign key references. [#127935][#127935]
```

> - It is now possible to assign to an element of a composite typed varialbe in [PL/pgSQL]({% link v24.3/plpgsql.md %}). For example, given a variable `foo` with two integer elements `x` and `y`, the following assignment statement is allowed: `foo.x := 100;`. [#132628][#132628]

Reviewer: typo: s/varialbe/variable/

> - It is now possible to assign to an element of a composite typed varialbe in [PL/pgSQL]({% link v24.3/plpgsql.md %}). For example, given a variable `foo` with two integer elements `x` and `y`, the following assignment statement is allowed: `foo.x := 100;`. [#132628][#132628]

Author: ```suggestion
- It is now possible to assign to an element of a composite typed variable in [PL/pgSQL]({% link v24.3/plpgsql.md %}). For example, given a variable `foo` with two integer elements `x` and `y`, the following assignment statement is allowed: `foo.x := 100;`. [#132628][#132628]
```

- Fixed a bug existing since before v23.1 that could lead to incorrect results in rare cases. The bug requires a lookup [join]({% link v24.3/joins.md %}) into a table with a computed [index]({% link v24.3/indexes.md %}) column, where the [computed column]({% link v24.3/computed-columns.md %}) expression is composite sensitive. A composite sensitive expression can compare differently if supplied non-identical but equivalent input values (e.g., `2.0::DECIMAL` versus `2.00::DECIMAL`). [#126345][#126345]
- Fixed a bug that caused quotes around the name of a routine to be dropped when it was called within another [routine]({% link v24.3/user-defined-functions.md %}). This could prevent the correct routine from being resolved if the nested routine name was case-sensitive. The bug has existed since v24.1 when nested routines were introduced. [#131643][#131643]
- Fixed a bug where the [SQL shell]({% link v24.3/cockroach-sql.md %}) would print out the previous error message when executing the `quit` command. [#130736][#130736]

> - Fixed a bug where a [span statistics]({% link v24.3/show-ranges.md %}#span-statistics) request on a mixed-version cluster resulted in an null pointer exception. [#132349][#132349]

Reviewer: typo: "an null pointer" -> "a null pointer"

> - Fixed a bug where a [span statistics]({% link v24.3/show-ranges.md %}#span-statistics) request on a mixed-version cluster resulted in an null pointer exception. [#132349][#132349]

Author: ```suggestion
- Fixed a bug where a [span statistics]({% link v24.3/show-ranges.md %}#span-statistics) request on a mixed-version cluster resulted in a null pointer exception. [#132349][#132349]
```
